railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -55,12 +57,14 @@ def parse_type_and_options(type)
         end
       end
 
-      def initialize(name, type=nil, index_type=false, attr_options={})
-        @name           = name
-        @type           = type || :string
-        @has_index      = INDEX_OPTIONS.include?(index_type)
-        @has_uniq_index = UNIQ_INDEX_OPTIONS.include?(index_type)
-        @attr_options   = attr_options
+      def initialize(name, type=nil, index_type=false, assigned_default=nil, attr_options={})
+        @name             = name
+        @type             = type || :string
+        @has_index        = INDEX_OPTIONS.include?(index_type)
+        @has_uniq_index   = UNIQ_INDEX_OPTIONS.include?(index_type)
+        @assigned_default = assigned_default
+        @has_default      = assigned_default.present? 
```

Comment:
Let's just put this in the method there's no reason to have it be an instance variable